src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,296 @@
+    this.files = this.files.filter((f: File) => !this._checkFileEquality(file, f));
+    this.fileChangedEvent.emit(this.files);
+  }
+
+  /** Calculates the correct unit for the file's size. */
+  private _formatFileSize(size: number): string {
+    const i: number = Math.floor(Math.log(size) / Math.log(1024));
+    return `${(size / Math.pow(1024, i)).toFixed(0)} ${this._suffixes[i]}`;
+  }
+
+  private _renderDefaultMode(): JSX.Element {
+    return (
+      <sbb-button
+        variant="secondary"
+        size="m"
+        is-static={true}
+        icon-name="folder-open-small"
+        disabled={this.disabled}
+        ref={(el): void => {
+          this._loadButton = el;
+        }}
+      >
+        {i18nFileSelectorButtonLabel[this._currentLanguage]}
+      </sbb-button>
+    );
+  }
+
+  private _renderDropzoneArea(): JSX.Element {
+    return (
+      <div class="sbb-file-selector__dropzone-area">
+        <div class="sbb-file-selector__dropzone-area--icon">
+          <sbb-icon name="folder-open-medium"></sbb-icon>
+        </div>
+        <div class="sbb-file-selector__dropzone-area--title">{this.titleContent}</div>
+        <div class="sbb-file-selector__dropzone-area--subtitle">
+          {i18nFileSelectorSubtitleLabel[this._currentLanguage]}
+        </div>
+        <div class="sbb-file-selector__dropzone-area--button">
+          <sbb-button
+            variant="secondary"
+            size="m"
+            disabled={this.disabled}
+            is-static={true}
+            ref={(el): void => {
+              this._loadButton = el;
+            }}
+          >
+            {i18nFileSelectorButtonLabel[this._currentLanguage]}
+          </sbb-button>
+        </div>
+      </div>
+    );
+  }
+
+  private _renderFileList(): JSX.Element {
+    return (
+      <ul class="sbb-file-selector__file-list" role="list">
+        {this.files.map((file: File) => (
+          <li class="sbb-file-selector__file">
+            <span class="sbb-file-selector__file-details">
+              <span class="sbb-file-selector__file-name">{file.name}</span>
+              <span class="sbb-file-selector__file-size">{this._formatFileSize(file.size)}</span>
+            </span>
+            <sbb-button
+              variant="secondary"
+              size="m"
+              icon-name="trash-small"
+              onClick={() => this._removeFile(file)}
+              aria-label={i18nFileSelectorDeleteFile[this._currentLanguage]}
+            ></sbb-button>
+          </li>
+        ))}
+      </ul>
+    );
+  }
+
+  public render(): JSX.Element {
+    return (
+      <div class="sbb-file-selector">
+        <div
+          class="sbb-file-selector__input-container"
+          onDragEnter={(e) => this._onDragEnter(e)}
+          onDragOver={(e) => this._blockEvent(e)}
+          onDragLeave={(e) => this._onDragLeave(e)}
+          onDrop={(e) => this._onFileDrop(e)}
+        >
+          <label htmlFor="sbb-file-selector__hidden-input">
```

Comment:
id reference should be necessary if nested inside label
```suggestion
          <label>
```

Reply:
Removed in [58e6011](https://github.com/lyne-design-system/lyne-components/pull/1894/commits/58e6011496b77e847dde243ffed0a60e45915757)